src/dockerconfig/docker_config.go

```diff
@@ -76,15 +59,36 @@ func (config *DockerConfig) SaveCustomCAs(
 	certs := &corev1.ConfigMap{}
 	if err := config.ApiReader.Get(ctx, client.ObjectKey{Namespace: config.Dynakube.Namespace, Name: config.Dynakube.Spec.TrustedCAs}, certs); err != nil {
 		log.Info("failed to load trusted CAs")
-		return err
+		return errors.WithStack(err)
 	}
 	if certs.Data[dtclient.CustomCertificatesConfigMapKey] == "" {
-		return fmt.Errorf("failed to extract certificate configmap field: missing field certs")
+		return errors.New("failed to extract certificate configmap field: missing field certs")
 	}
 	if err := fs.WriteFile(path, []byte(certs.Data[dtclient.CustomCertificatesConfigMapKey]), 0666); err != nil {
 		log.Info("failed to save custom certificates")
-		return err
+		return errors.WithStack(err)
 	}
 	config.TrustedCertsPath = path
 	return nil
 }
+
+func parseDockerAuthsFromSecret(secret *corev1.Secret) (map[string]DockerAuth, error) {
+	if secret == nil {
+		return nil, errors.New("given secret is nil")
```

Comment:
could this be a more specific error message ? could not generate docker auths from empty secret